libs/features/mas/web-components/src/cdt/countdown-timer.js

```diff
@@ -0,0 +1,119 @@
+    this.minutesLeft = '';
+    this.isVisible = false;
+  }
+
+  firstUpdated() {
+    this.#timeWords = this.daysHoursMins.split(' ');
+    this.#timeRangesEpoch = this.timeRanges.split(',');
+    this.countdownStart();
+  }
+
+  disconnectedCallback() {
+    super.disconnectedCallback();
+    clearInterval(this.countdownInterval);
+  }
+
+  countdownCompleted() {
+    this.isVisible = false;
+    clearInterval(this.countdownInterval);
+  }
+
+  countdownStart() {
+    this.countdownUpdate();
+    this.countdownInterval = setInterval(() => {
+      this.countdownUpdate();
+    }, 60000);
+  }
+
+  countdownUpdate() {
+
+    let currentTime = Date.now();    
+    //find the time range where the current time is between the start and end time
+    for (let i = 0; i < this.#timeRangesEpoch.length; i += 2) {
+      const startTime = this.#timeRangesEpoch[i];
+      const endTime = this.#timeRangesEpoch[i + 1];
+      
+      if (currentTime >= startTime && currentTime <= endTime) {
+        this.isVisible = true;
+        let diffTime = endTime - currentTime;
+        this.daysLeft = Math.floor(diffTime / (1000 * 60 * 60 * 24));
+        this.hoursLeft = Math.floor((diffTime % (1000 * 60 * 60 * 24)) / (1000 * 60 * 60));
+        this.minutesLeft = Math.floor((diffTime % (1000 * 60 * 60)) / (1000 * 60));
+        this.requestUpdate(); 
+        return;
+      }
+    }
+
+    //if the current time is not in any of the time ranges, the countdown is completed
+    this.countdownCompleted();    
+    this.requestUpdate(); 
+  }
+
+  render() {
+    if (!this.isVisible) {
+      return html``;
+    } else {
+      return html`
+      <div class="${this.classList}">
+        <div class="timer-label">${this.label}</div>
+        <div class="timer-container-parent">
+          <div class="timer-container">
+              <div class="timer-unit-container">
+                <div class="timer-box">${Math.floor(this.daysLeft/10)}</div>
+                <div class="timer-box">${this.daysLeft%10}</div>
+              </div>
+              <div class="timer-unit-label">${this.#timeWords[0]}</div>
+          </div>
+          <div class="timer-label">:</div>
+          <div class="timer-container">
+              <div class="timer-unit-container">
+                <div class="timer-box">${Math.floor(this.hoursLeft/10)}</div>
+                <div class="timer-box">${this.hoursLeft%10}</div>
+              </div>
+              <div class="timer-unit-label">${this.#timeWords[1]}</div>
+          </div>
+          <div class="timer-label">:</div>
+          <div class="timer-container">
+              <div class="timer-unit-container">
+                <div class="timer-box">${Math.floor(this.minutesLeft/10)}</div>
+                <div class="timer-box">${this.minutesLeft%10}</div>
+              </div>
+              <div class="timer-unit-label">${this.#timeWords[2]}</div>
+          </div>
+        </div>
+    </div>`;
+    }
+  }
+
+  lpad(input, length) {
```

Comment:
if this is a util function, you can declare it outside of your class,  otherwise it becomes a function of your web component in the DOM.

Reply:
removed as unused